common/Log.cpp

```diff
@@ -665,14 +668,23 @@ namespace Log
                 &GenericLogger::get(Static.getInited() ? Static.getName() : std::string()));
     }
 
-    bool isEnabled(Level l)
+    bool isEnabled(Level l, Area a)
     {
         if (IsShutdown)
             return false;
 
         Log::Level logLevel = GenericLogger::mapToLevel(
             static_cast<Poco::Message::Priority>(logger().getLevel()));
-        return logLevel >= static_cast<int>(l);
+
+        if (logLevel < static_cast<int>(l))
+            return false;
+
+        bool disabled = AreasDisabled[static_cast<size_t>(a)];
+
+        // Areas shouldn't disable warnings & errors
+        assert(!disabled || logLevel > static_cast<int>(Level::WRN));
```

Comment:
seeing as this will be configurable from coolwsd.xml do we want end users with an NDEBUG build to be able to disable areas while using a warning/error debuglevel?